syntax = "proto3";

package core;

service Drone {
  rpc ReceiveFile(stream FileFragment) returns (Status);
  rpc SendFiles(FileRequest) returns (Status);
}

message Status {
  string message = 1;
  int32 statusCode = 2;
}

message FileFragment {
  string filename = 1;
  int32 fragmentId = 2;
  bytes fragmentContent = 3;
}

message FileRequestPayload {
  string filePath = 1;
  int32 fragmentSize = 2;
}

message FileRequest {
  repeated FileRequestPayload fileRequestPayloads = 1;
}
